osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyMainCirclePiece.cs

```diff
@@ -107,37 +105,10 @@ private void load()
             if (overlayAboveNumber)
                 OverlayLayer.ChangeChildDepth(hitCircleOverlay, float.MinValue);
 
-            accentColour.BindTo(drawableObject.AccentColour);
-            indexInCurrentCombo.BindTo(drawableOsuObject.IndexInCurrentComboBindable);
-
-            Texture getTextureWithFallback(string name)
-            {
-                Texture tex = null;
-
-                if (!string.IsNullOrEmpty(priorityLookup))
-                {
-                    tex = skin.GetTexture($"{priorityLookup}{name}");
-
-                    if (!allowFallback)
-                        return tex;
-                }
-
-                return tex ?? skin.GetTexture($"hitcircle{name}");
-            }
-
-            Drawable getAnimationWithFallback(string name, double frameLength)
+            if (drawableOsuObject != null)
```

Comment:
Tying into the comment above, I believe that this has no right to occur in any real non-broken usage. I'd argue that it's better to let this loudly crash rather than silently not work, and therefore there should be a hard cast somewhere to non-nullable to elide all of the nullchecks.